core/mavlink_parameters.cpp

```diff
@@ -38,10 +38,10 @@ void MAVLinkParameters::set_param_async(const std::string &name,
     //     LogDebug() << "setting param " << name << " to " << value.get_int();
     // }
 
-    if (name.size() > PARAM_ID_LEN) {
+    if (name.size() >= PARAM_ID_LEN) {
```

Comment:
Was that a bug before?

Reply:
Only if you tried setting/getting a param longer than allowed.

MAVLink params are allowed to have a name of 16 chars + termination character.

The thing is that `PARAM_ID_LEN` is 17, so if `name.size()` is 17 it's already too long because `size()` does not include the termination character.

Reply:
> Only if you tried setting/getting a param longer than allowed.

:rofl: 

Got it, thanks for the explanation!

Reply:
> MAVLink params are allowed to have a name of 16 chars + termination character.

Not quite, they are sent with null character if less than 16 chars, but are not sent with termination char at length 16 chars. https://mavlink.io/en/messages/common.html#PARAM_VALUE